R-package/src/cmake/modules/FindLibR.cmake

```diff
@@ -174,17 +174,28 @@ execute_process(
   OUTPUT_VARIABLE LIBR_LIB_DIR
 )
 
+set(LIBR_HOME ${LIBR_HOME} CACHE PATH "R home directory")
+set(LIBR_EXECUTABLE ${LIBR_EXECUTABLE} CACHE PATH "R executable")
+set(LIBR_INCLUDE_DIRS ${LIBR_INCLUDE_DIRS} CACHE PATH "R include directory")
+set(LIBR_LIB_DIR ${LIBR_LIB_DIR} CACHE PATH "R shared libraries directory")
+
 # look for the core R library
 find_library(
   LIBR_CORE_LIBRARY
-  NAMES R
-  HINTS "${CMAKE_CURRENT_BINARY_DIR}" "${LIBR_LIB_DIR}" "${LIBR_HOME}/bin" "${LIBR_LIBRARIES}"
+  NAMES R R.dll
+  HINTS "${CMAKE_CURRENT_BINARY_DIR}" "${LIBR_HOME}/lib" "${LIBR_HOME}/bin/${R_ARCH}" "${LIBR_HOME}/bin" "${LIBR_LIBRARIES}"
```

Comment:
Please save `HINTS` into a variable to not list them twice in `find_library` and `find_file`.

Reply:
oh ok sure

Reply:
ah I forgot to do this in recent commits. Added in https://github.com/microsoft/LightGBM/pull/2936/commits/cdf1df8696846b86853d1858996e0211805d0eed